owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/security/PassCodeViewModel.kt

```diff
@@ -64,6 +70,25 @@ class PassCodeViewModel(
     fun setMigrationRequired(required: Boolean) =
         preferencesProvider.putBoolean(PassCodeActivity.PREFERENCE_MIGRATION_REQUIRED, required)
 
+    fun setLastUnlockTimestamp() =
+        preferencesProvider.putLong(PREFERENCE_LAST_UNLOCK_TIMESTAMP, SystemClock.elapsedRealtime())
+
+    fun getNumberOfAttempts() = preferencesProvider.getInt(PREFERENCE_LOCK_ATTEMPTS, 0)
+
+    fun increaseNumberOfAttempts() {
+        preferencesProvider.putInt(PREFERENCE_LOCK_ATTEMPTS, getNumberOfAttempts().plus(1))
+        preferencesProvider.putLong(PREFERENCE_LAST_UNLOCK_ATTEMPT_TIMESTAMP, SystemClock.elapsedRealtime())
+    }
+
+    fun resetNumberOfAttempts() =
+        preferencesProvider.putInt(PREFERENCE_LOCK_ATTEMPTS, 0)
+
+    fun getTimeToUnlockLeft(): Long {
+        val timeLocked = 1.5.pow(getNumberOfAttempts()).toLong() * 1000
```

Comment:
Try using Kotlin's binary operators instead of expressions, you can check it here: https://kotlinlang.org/docs/operator-overloading.html#binary-operations

Reply:
👍